rust/src/adapter/datadog.rs

```diff
@@ -0,0 +1,160 @@
+use crate::{
+    adapter::datadog_formatter::DatadogFormatter,
+    Event, Metadata,
+};
+use std::{
+    sync::{
+        atomic::{AtomicUsize, Ordering},
+        Arc,
+    }, collections::HashMap
+};
+use serde_json::json;
+use tokio::sync::Mutex;
+use ureq;
+use url::Url;
+
+use super::{
+    datadog_formatter::{self, Trace, Span},
+    Adapter, Collector,
+};
+
+pub type DatadogAdapterContainer = Arc<Mutex<DatadogAdapter>>;
+
+#[derive(Clone)]
+pub struct DatadogAdapter {
+    pub trace_id: u64,
+    pub spans: Vec<Span>,
+    pub config: DatadogConfig,
+}
+
+#[derive(Clone)]
+pub struct DatadogConfig {
+    pub agent_host: String,
+    pub service_name: String,
+    pub default_tags: HashMap<String, String>,
+    pub trace_type: String,
+}
+
+impl DatadogConfig {
+    pub fn new() -> DatadogConfig {
+        DatadogConfig {
+            agent_host: "http://localhost:8126".into(),
```

Comment:
TODO Should we also check env variables? there are some well known ddog env variables set for this kind of configs.

Reply:
Yea I think we should probably do that in these vendor specific adapters. Would be nice to just drop in the SDK and things work along with the rest of your app.